### PR TITLE
Implement ListPaymentsHistory

### PR DESCRIPTION
**Pull request by amackillop**
This is the API and implementation for the ListPaymentsHistory endpoint

### CODE REVIEW

**Review by saidalaoui-sq — APPROVED** [review]
Pending rebase once build fix included in [PR](https://github.com/tnull/ldk-node-hack-server/pull/28) is merged. LGTM otherwise

File: server/src/service.rs [hunk 1]
```diff
@@ -50,6 +52,98 @@ impl NodeService {
...
+					}
+				},
+			}
+		}
+		let payments = self.node.list_payments();
+		let msg = protos::PaymentsHistoryResponse {
+			payments: payments
+				.iter()
+				.map(|payment| protos::PaymentDetails {
+					id: Some(protos::PaymentId { data: payment.id.0.to_vec() }),
```
[the remark this answers is not in the document]
Comment (amackillop) [reply server/src/service.rs:128]: PaymentId is missing a Display implementation. Should we add that to ldk-node?

File: server/src/service.rs [hunk 2]
```diff
@@ -62,6 +64,98 @@ impl NodeService {
 		make_response(msg.encode_to_vec())
 	}
 
+	fn handle_get_payment_history_request(
+		&self, _: Req,
+	) -> <NodeService as Service<Request<Incoming>>>::Future {
+		fn to_proto(kind: &PaymentKind) -> protos::PaymentKind {
```
Comment (saidalaoui-sq) [inline server/src/service.rs:70]: Let's extract this to be reused for getting payment details

**Review by saidalaoui-sq — APPROVED** [review]

**Comment by amackillop** [comment]
closes #20

**Pull request merged**